compiler/rustc_ast_pretty/src/pprust/state.rs

```diff
@@ -469,14 +469,22 @@ pub trait PrintState<'a>: std::ops::Deref<Target = pp::Printer> + std::ops::Dere
                 true,
                 span,
             ),
-            MacArgs::Empty | MacArgs::Eq(..) => {
+            MacArgs::Empty => {
                 self.print_path(&item.path, false, 0);
-                if let MacArgs::Eq(_, token) = &item.args {
-                    self.space();
-                    self.word_space("=");
-                    let token_str = self.token_to_string_ext(token, true);
-                    self.word(token_str);
-                }
+            }
+            MacArgs::Eq(_, MacArgsEqKind::Ast(expr)) => {
+                self.print_path(&item.path, false, 0);
+                self.space();
+                self.word_space("=");
+                let token_str = self.expr_to_string(expr);
+                self.word(token_str);
+            }
+            MacArgs::Eq(_, MacArgsEqKind::Hir(lit)) => {
```

Comment:
This case shouldn't be reachable if all `MacArgsEq::Hir`s are created during lowering.

Reply:
Without it, I get two rustdoc test failures"
```
failures:
    [rustdoc] src/test/rustdoc/attribute-rendering.rs
    [rustdoc] src/test/rustdoc/attributes.rs
```

Reply:
Ah, I see, this code is reused by HIR pretty-printing too, despite being in `rustc_ast_pretty`.